sbncode/CAFMaker/FillReco.cxx

```diff
@@ -738,5 +759,11 @@ namespace caf
     return;
   }
 
+  //......................................................................
+  template<class T, class U>
+  void CopyPropertyIfSet( const std::map<std::string, T> map, const std::string search, U &value )
+  {
+    if (map.count(search)) value = map.at(search);
+  }
```

Comment:
`search` should definitely be a const reference. I think `map` (bad name because it clashes with the type, maybe `props`?) should be too, but I think you will need to change the implementation to not use `at()`. You can use `find()` to get an iterator and dereference that.

Something like this
```suggestion
  //......................................................................
  template<class T, class U>
  void CopyPropertyIfSet( const std::map<std::string, T>& props, const std::string& search, U& value )
  {
    auto it = props.find(search);
    if(it != props.end()) value = *it;
  }
```

Reply:
The map iterator returns a pair, so I think I want `it->second` instead of `*it` right? Update to method and removing the include noted below now committed. Does this seem better?